### PR TITLE
Add more new sniffs

### PR DESCRIPTION
**Pull request by Majkl578**
Second part of new sniffs, this time not 7.1-related, but more generic instead. Comments for each sniff are in the ruleset.
Most of these should already be ok on 99% of code base and are mostly to maintain consistence.
Also removed sniffs for _spaces around `!`_, replaced by generic sniff for _spaces after `!`_ (as agreed in past and suggested in #9).

cc @alcaeus @lcobucci

### CODE REVIEW

File: lib/Doctrine/ruleset.xml [hunk 1]
```diff
@@ -3,31 +3,180 @@
...
+        <properties>
+            <property name="allowMultiline" value="true"/>
+        </properties>
+    </rule>
+    <!-- Force scope changes to increase indentation -->
+    <rule ref="Generic.WhiteSpace.ScopeIndent"/>
+    <!-- Require EOL at EOF -->
+    <rule ref="Generic.Files.EndFileNewline"/>
+    <!-- Forbid multiple classes per file -->
+    <rule ref="Generic.Files.OneClassPerFile"/>
```
Comment (Majkl578) [inline lib/Doctrine/ruleset.xml]: This one will be problematic for ORM where tests have their entities defined in the same file. Could be excluded on project level though.

Comment (Ocramius) [reply lib/Doctrine/ruleset.xml]: Should be excluded per-directory IMO

Comment (Majkl578) [reply lib/Doctrine/ruleset.xml]: If that is possible to configure, even better. 👍

Comment (Ocramius) [reply lib/Doctrine/ruleset.xml]: It would simply require a custom local `phpcs.xml` file per directory to be inspected, with overrides for disabled rules

Comment (lcobucci) [reply lib/Doctrine/ruleset.xml]: Isn't this covered by `PSR1.Classes.ClassDeclaration.MultipleClasses` (which is part of PSR-2)? If so, we already exclude things in the ORM

https://github.com/doctrine/doctrine2/blob/374e7ace49d864dad8cddbc55346447c8a6a2083/phpcs.xml.dist#L20-L22

Comment (Majkl578) [reply lib/Doctrine/ruleset.xml]: Didn't realize it's already included 👍, we can remove these checks then, PSR-1 sniff provides check for classes/interfaces/traits at once.

Comment (Majkl578) [reply lib/Doctrine/ruleset.xml]: Removed in 2ac121d.

**Pull request merged**